### PR TITLE
Camera Autoname change

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request
Makes autonaming cameras happen based on a preset variable rather than hardcode.

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
code: Security camera autoname is now determined by a variable rather than hardcode.
/:cl:

### CODE REVIEW

File: code/game/machinery/camera/camera.dm [hunk 1]
```diff
@@ -58,6 +62,19 @@
 	set_pixel_location()
 	update_icon()
 
+	//This camera automatically sets it's name to whatever the area that it's in is called.
+	if(autoname)
+		autonumber = 1
+		var/area/A = get_area(src)
```
Comment (harryob) [inline code/game/machinery/camera/camera.dm:68]:
```suggestion
		var/area/my_area = get_area(src)
```

File: code/game/machinery/camera/camera.dm [hunk 2]
```diff
@@ -58,6 +62,19 @@
...
+		if(A)
+			for(var/obj/structure/machinery/camera/autoname/C in GLOB.machines)
+				if(C == src) continue
+				var/area/CA = get_area(C)
+				if(CA.type == A.type)
+					if(C.autonumber)
+						autonumber = max(autonumber, C.autonumber+1)
```
Comment (harryob) [inline code/game/machinery/camera/camera.dm:75]:
```suggestion
		if(my_area)
			for(var/obj/structure/machinery/camera/autoname/current_camera in GLOB.machines)
				if(current_camera == src) 
					continue
				var/area/current_camera_area = get_area(current_camera)
				if(current_camera_area.type != my_area.type)
					continue

				if(!current_camera.autonumber)
					continue

				autonumber = max(autonumber, current_camera.autonumber + 1)
```

File: code/game/machinery/camera/camera.dm [hunk 3]
```diff
@@ -58,6 +62,19 @@
...
+			c_tag = "[A.name] #[autonumber]"
```
Comment (harryob) [inline code/game/machinery/camera/camera.dm:76]:
```suggestion
			c_tag = "[my_area.name] #[autonumber]"
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by Birdtalon — APPROVED** [review]

**Pull request merged**